src/entities/drawer/ui/DrawingBoard.tsx

```diff
@@ -47,137 +23,49 @@ type Props = {
-    drawingPathRef.current = null;
-    canvasRef.current?.clear(Skia.Color('transparent'));
-  };
-
-  const getNow = (): number => new Date().getTime();
-
-  const createValuePoint = (point: Point): DrawPoint => {
-    const logPoint = { ...point, time: getNow() };
-
-    const liveEventLogPoint = {
-      x: (point.x / width) * 100,
-      y: (point.y / width) * 100,
-      time: getNow(),
-    };
+  const vector = width / 100;
 
-    onLog(liveEventLogPoint);
+  const callbacksRef = useRef({
+    onStarted,
+    onLog,
+  });
 
-    return logPoint;
+  callbacksRef.current = {
+    onStarted,
+    onLog,
   };
 
-  const createValueLine = ({ x, y }: Point): void => {
-    const logPoint = { x, y, time: getNow() };
-    const liveEventLogPoint = {
-      x: (x / width) * 100,
-      y: (y / width) * 100,
-      time: getNow(),
-    };
+  const canvasRef = useRef<SketchCanvas | null>();
 
-    onLog(liveEventLogPoint);
-
-    drawingValueLineRef.current = {
-      startTime: getNow(),
-      points: [logPoint],
-    };
-  };
-
-  const addValuePoint = (point: DrawPoint) => {
-    getValueLine()!.points.push(point);
-  };
-
-  const getValueLine = (): DrawLine | null => {
-    return drawingValueLineRef.current;
-  };
-
-  const isEqualToLastPoint = (point: Point): boolean => {
-    const line = getValueLine()!;
-
-    if (!line.points.length) {
-      return false;
-    }
-
-    const lastPoint: DrawPoint = line.points.slice(-1)[0];
-
-    return (
-      Math.round(point.x) === Math.round(lastPoint.x) &&
-      Math.round(point.y) === Math.round(lastPoint.y)
-    );
-  };
+  const linesRef = useRef<Array<DrawLine>>(value);
 
-  const onTouchStart = (touchInfo: TouchInfo) => {
-    bezierCache.splice(0, bezierCache.length);
+  const lineTracker = useMemo(() => {
+    return new LinesTracker(linesRef, {
+      onPointAdded: point => {
+        callbacksRef.current.onLog(point.scale(vector));
+      },
+    });
+  }, [vector]);
 
-    resetCurrentLine();
-
-    const point: Point = { x: touchInfo.x, y: touchInfo.y };
-
-    createValueLine(point);
-    drawPath();
-    onStarted();
-  };
-
-  const onTouchProgress = (touchInfo: TouchInfo) => {
-    const point: Point = { x: touchInfo.x, y: touchInfo.y };
-
-    if (isEqualToLastPoint(point)) {
-      drawPath();
-      return;
-    }
-
-    addValuePoint(createValuePoint(point));
-    drawPath();
-  };
-
-  const onTouchEnd = () => {
-    if (!drawingValueLineRef.current) {
-      return;
-    }
+  const onStrokeStart = useCallback(
+    (x: number, y: number) => {
+      lineTracker.createLine(x, y);
+    },
+    [lineTracker],
+  );
 
-    const newLine = { ...drawingValueLineRef.current };
+  const onStrokeChanged = useCallback(
+    (x: number, y: number) => {
+      lineTracker.progressLine(x, y);
+    },
+    [lineTracker],
+  );
 
-    const lines = [...value, newLine];
+  const onStrokeEnd = useCallback(() => {
+    const lines = linesRef.current;
 
```

Comment:
Please check export drawer data in Admin (If you didn't). Just to veryfy that output is similar to the previous version.